### PR TITLE
DDF-4388 Generate source jars on release

### PR DESCRIPTION
**Pull request by rymach**
#### What does this PR do?
Generate source jars on release
Address some inspection warnings 
#### Who is reviewing it? 
@coyotesqrl 
@paouelle 
@bakejeyner 
@mcalcote 
#### Select relevant component teams: 
@codice/build 
#### Ask 2 committers to review/merge the PR and tag them here.

@bdeining
@vinamartin
#### How should this be tested?
CI build
#### What are the relevant tickets?
[DDF-4388](https://codice.atlassian.net/browse/DDF-4388)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/transformer/catalog-transformer-attribute/pom.xml [hunk 1]
```diff
@@ -19,7 +19,6 @@
         <artifactId>transformer</artifactId>
         <version>2.14.0-SNAPSHOT</version>
     </parent>
-    <groupId>ddf.catalog.transformer</groupId>
```
Comment (paouelle) [inline catalog/transformer/catalog-transformer-attribute/pom.xml]: ❓ Why remove the group id? It is clearer if we keep it. This applies to all subsequent pom files.

File: gitsetup/pom.xml [hunk 2]
```diff
@@ -23,7 +23,6 @@
         <version>2.14.0-SNAPSHOT</version>
     </parent>
 
-    <groupId>ddf</groupId>
```
Comment (paouelle) [inline gitsetup/pom.xml]: ❓ Why remove the group id? It is clearer if we keep it. This one might actually cause a problem when running the githooks.

File: pom.xml [hunk 3]
```diff
@@ -16,7 +16,6 @@
          xmlns="http://maven.apache.org/POM/4.0.0"
          xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
     <modelVersion>4.0.0</modelVersion>
-    <groupId>ddf</groupId>
```
Comment (paouelle) [inline pom.xml]: ❗️ I don't think we should be removing it in the parent pom.

File: catalog/transformer/catalog-transformer-attribute/pom.xml (hunk 1 above)
Comment (rymach) [reply catalog/transformer/catalog-transformer-attribute/pom.xml]: Definition of groupId is redundant, because it's inherited from the parent

Comment (paouelle) [reply catalog/transformer/catalog-transformer-attribute/pom.xml]: I know it is. However, the fact is that it is much clearer if it is present.

Comment (rymach) [reply catalog/transformer/catalog-transformer-attribute/pom.xml]: Inheriting the groupId follows the same paradigm as all the other pom properties we inherit and has the same benefits, like improved maintainability. 

Are you ok with adding `<!--suppress MavenRedundantGroupId -->` if we want to keep these groupIds?

Comment (paouelle) [reply catalog/transformer/catalog-transformer-attribute/pom.xml]: I wouldn't. It doesn't provide any valid info. I have the same comment about all the other comments you added. I feel this PR is doing a lot more than intended and some of that should probably be discussed first.

Although it is inherited, when you are looking for the group id, you have to manually traverse all in search of it. In addition! Someone might not realize that changing the group id on a pom would now affect all children poms. Properties are typically defined only in the top pom and should never be defined in other poms. It just keep maintenance more complex and makes it easier to do mistakes.

**Pull request closed without merging**